02_bubble/CameronSchafer/bubble.go

```diff
@@ -0,0 +1,95 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func bubble(s []int) []int {
+	check := false
+
+	for !check {
+		tempCheck := true
+		//sort once and then check
+		for i := 0; i < len(s); i++ {
+			//check if in last slice
+			if i+1 < len(s) {
+				if s[i] > s[i+1] {
+					//swap the elems around
+					s[i], s[i+1] = s[i+1], s[i]
+					tempCheck = false
+				}
+			}
+		}
+		check = tempCheck
+	}
+	return s
```

Comment:
Consider refactoring to return a copy of slice.

Reply:
Changed this function to sort a copy of the slice.